routing/async_router.cpp

```diff
@@ -371,17 +371,21 @@ void AsyncRouter::CalculateRoute()
                           [delegateProxy, route, code]() { delegateProxy->OnReady(route, code); });
   }
 
-  bool const needAbsentRegions = (code != RouterResultCode::Cancelled);
+  bool const needAbsentRegions = (code != RouterResultCode::Cancelled &&
+          route->GetRouterId() != "helicopter-router");
```

Comment:
That line is not called too often, so should be ok. I would also try to reuse a router-id constexpr constant instead of copy-pasting a string. Or maybe add a method like route->IsHelicopterRouter()